desktop/cmp/grid/impl/filter/ColumnHeaderFilterModel.js

```diff
@@ -38,15 +38,20 @@ export class ColumnHeaderFilterModel extends HoistModel {
         return this.gridFilterModel.getColumnFilters(this.field);
     }
 
+    @computed.struct
+    get pendingFilter() {
+        const {activeTabId} = this.tabContainerModel;
```

Comment:
does this yield an immediate filter change just from changing tabs?   That seems surprising.

Reply:
Good point! :|

Not sure what the best approach is here. I could tweak the logic to wait until you've **made a change** in the new tab to commit the filter?

Alternatively, this could be a good justification to make require the user to explicitly clear out their filter upon switching tabs, which we currently only do when switching to the Values Tab from the Custom Tab.

Reply:
I think we could just  flip things around so that each UI tab puts on its own reaction, such that *when it is visible*, and its value changes, then it should do the commit action

Reply:
So playing with this.  Its really nice when the data set is the right size, and its all snappy.  

One issue I have is that for the custom tab, its now updating on every keystroke.  This to me seems overly aggressive.  Its especially problematic when you are trying to type in, say, a number, and you get intermediate results based on the first digits that you type.

Think we could potentially make the text fields on the Custom tab commit on change: false, so you had to hit enter, or leave the field for the new value to flow through.  (f we did that, would also probably need to always enable the apply button as well).

I also see a flash in the FilterChooser (in our toolbox example tab) whenever I update the filter via the single keystrokes.  Not sure what that is, but its annoying.  We might address the underlying cause, but that will be less important if we change the behavior as described above.

Reply:
Could be good to add a little switch at the bottom of the toolbox example just to turn this `commitOnChange` behavior on and off